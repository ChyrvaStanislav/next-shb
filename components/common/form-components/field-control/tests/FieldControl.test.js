import React from 'react';
import { render } from '@testing-library/react';
import FieldControl from '../index';

const Component = props => (
  <FieldControl
    formId="Form Id"
    field={{
      name: 'someName'
    }}
    form={{
      touched: {
        someName: false,
      },
      errors: {
        someName: '',
      }
    }}
    className="someClass"
    label="Some label"
    {...props}
  >
    <input name="children" value="children" onChange={() => {}} />
  </FieldControl>
);

describe('FieldControl renders', () => {
  it('should FieldControl renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });
});


describe('Error message', () => {
  it('should Error message shows', () => {
    const { getByText } = render(<Component form={{
      touched: {
        someName: true,
      },
      errors: {
        someName: 'ERROR',
      }
    }}
    />);

    const errorMessage = getByText('ERROR');
    expect(errorMessage).toBeInTheDocument();
  });
});
