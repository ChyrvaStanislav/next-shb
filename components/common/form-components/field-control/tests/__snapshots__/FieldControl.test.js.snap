// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldControl renders should FieldControl renders 1`] = `
<DocumentFragment>
  <section
    class="textareaContainer someClass"
  >
    <label
      class="small_filters label"
      for="Form Id-someName"
      other=""
    >
      Some label
    </label>
    <input
      name="children"
      value="children"
    />
  </section>
</DocumentFragment>
`;
